### ISSUE TITLE
Theme not found when using Flow alpha 17

### ISSUE THREAD

**Issue by Artur-** [vaadin/flow#18894]
### Description of the bug

When trying to get the theme folder using `ThemeUtils.getThemeName` then it tries to use `src/main/frontend` even when the theme is in `frontend/themes` and fails to find the theme name.

Potentially related to the fact that the code is called through the push channel(?)

```
	at com.vaadin.copilot.plugins.themeeditor.ThemeModifier.lambda$getThemeFolder$3(ThemeModifier.java:206) ~[classes/:na]
	at java.base/java.util.Optional.orElseThrow(Optional.java:403) ~[na:na]
	at com.vaadin.copilot.plugins.themeeditor.ThemeModifier.getThemeFolder(ThemeModifier.java:205) ~[classes/:na]
	at com.vaadin.copilot.plugins.themeeditor.ThemeModifier.<init>(ThemeModifier.java:58) ~[classes/:na]
	at com.vaadin.copilot.plugins.themeeditor.ThemeEditorMessageHandler.<init>(ThemeEditorMessageHandler.java:60) ~[classes/:na]
	at com.vaadin.copilot.Copilot.setupCommands(Copilot.java:75) ~[classes/:na]
	at com.vaadin.copilot.Copilot.handleConnect(Copilot.java:64) ~[classes/:na]
	at com.vaadin.base.devserver.DebugWindowConnection.handleConnect(DebugWindowConnection.java:222) ~[vaadin-dev-server-24.4.0.alpha17.jar:na]
	at com.vaadin.base.devserver.DebugWindowConnection.onConnect(DebugWindowConnection.java:202) ~[vaadin-dev-server-24.4.0.alpha17.jar:na]
	at com.vaadin.flow.server.communication.PushHandler.lambda$onConnect$3(PushHandler.java:599) ~[flow-server-24.4.0.alpha17.jar:24.4.0.alpha17]
	at java.base/java.util.Optional.ifPresent(Optional.java:178) ~[na:na]
	at com.vaadin.flow.server.communication.PushHandler.lambda$onConnect$4(PushHandler.java:598) ~[flow-server-24.4.0.alpha17.jar:24.4.0.alpha17]
	at com.vaadin.flow.server.communication.PushHandler.callWithServiceAndSession(PushHandler.java:227) ~[flow-server-24.4.0.alpha17.jar:24.4.0.alpha17]
	at com.vaadin.flow.server.communication.PushHandler.onConnect(PushHandler.java:595) ~[flow-server-24.4.0.alpha17.jar:24.4.0.alpha17]
	at com.vaadin.flow.server.communication.PushAtmosphereHandler.onConnect(PushAtmosphereHandler.java:103) ~[flow-server-24.4.0.alpha17.jar:24.4.0.alpha17]
	at com.vaadin.flow.server.communication.PushAtmosphereHandler.onRequest(PushAtmosphereHandler.java:77) ~[flow-server-24.4.0.alpha17.jar:24.4.0.alpha17]
	at org.atmosphere.cpr.AsynchronousProcessor.action(AsynchronousProcessor.java:217) ~[atmosphere-runtime-3.0.3.slf4jvaadin2.jar:3.0.3.slf4jvaadin2]
	at org.atmosphere.cpr.AsynchronousProcessor.suspended(AsynchronousProcessor.java:103) ~[atmosphere-runtime-3.0.3.slf4jvaadin2.jar:3.0.3.slf4jvaadin2]
	at org.atmosphere.container.Servlet30CometSupport.service(Servlet30CometSupport.java:67) ~[atmosphere-runtime-3.0.3.slf4jvaadin2.jar:3.0.3.slf4jvaadin2]
	at org.atmosphere.cpr.AtmosphereFramework.doCometSupport(AtmosphereFramework.java:2284) ~[atmosphere-runtime-3.0.3.slf4jvaadin2.jar:3.0.3.slf4jvaadin2]
	at org.atmosphere.websocket.DefaultWebSocketProcessor.dispatch(DefaultWebSocketProcessor.java:574) ~[atmosphere-runtime-3.0.3.slf4jvaadin2.jar:3.0.3.slf4jvaadin2]
	at org.atmosphere.websocket.DefaultWebSocketProcessor.open(DefaultWebSocketProcessor.java:213) ~[atmosphere-runtime-3.0.3.slf4jvaadin2.jar:3.0.3.slf4jvaadin2]
	at org.atmosphere.container.JSR356Endpoint.onOpen(JSR356Endpoint.java:254) ~[atmosphere-runtime-3.0.3.slf4jvaadin2.jar:3.0.3.slf4jvaadin2]
	at org.apache.tomcat.websocket.server.WsHttpUpgradeHandler.init(WsHttpUpgradeHandler.java:131) ~[tomcat-embed-websocket-10.1.19.jar:10.1.19]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:938) ~[tomcat-embed-core-10.1.19.jar:10.1.19]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1744) ~[tomcat-embed-core-10.1.19.jar:10.1.19]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) ~[tomcat-embed-core-10.1.19.jar:10.1.19]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) ~[tomcat-embed-core-10.1.19.jar:10.1.19]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) ~[tomcat-embed-core-10.1.19.jar:10.1.19]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63) ~[tomcat-embed-core-10.1.19.jar:10.1.19]
	at java.base/java.lang.Thread.run(Thread.java:1589) ~[na:na]
```

### Expected behavior

It returns the theme name

### Minimal reproducible example

Flow test app in copilot

### Versions

- Vaadin / Flow version: 24.4 alpha17

**Comment by mshabarov** [comment]
`ThemeUtils` uses 
```
String frontendFolderPath = config.getStringProperty(
                    FrontendUtils.PARAM_FRONTEND_DIR,
                    FrontendUtils.DEFAULT_FRONTEND_DIR);
```
which doesn't take into account legacy location.

We have to find all similar places in the codes and fix them.

**Comment by Artur-** [comment]
Fixed in #18902
